internal/service/thirdpartyintegration/data_source_third_party_integrations.go

```diff
@@ -97,11 +73,11 @@ func integrationToSchema(d *schema.ResourceData, integration *matlas.ThirdPartyI
 		"service_key":                 integrationSchema.ServiceKey,
 		"team_name":                   integration.TeamName,
 		"channel_name":                integration.ChannelName,
-		"routing_key":                 integrationSchema.RoutingKey,
+		"routing_key":                 integration.RoutingKey,
 		"url":                         integrationSchema.URL,
 		"secret":                      integrationSchema.Secret,
 		"microsoft_teams_webhook_url": integrationSchema.MicrosoftTeamsWebhookURL,
-		"user_name":                   integrationSchema.UserName,
+		"user_name":                   integration.UserName,
```

Comment:
both of these attributes are returned by the API, so the values for the response (`integration`) is used.